src/v3/src/transformer/i18n/transformPasscodeHint.ts

```diff
@@ -35,8 +35,8 @@ export const transformPasscodeHint: TransformStepFnWithOptions = ({
       const { nextStep: { name } = {} } = transaction;
       const authenticatorKey = getAuthenticatorKey(transaction);
       if (!(name === IDX_STEP.IDENTIFY
-        || (name !== IDX_STEP.CHALLENGE_AUTHENTICATOR
-          && authenticatorKey !== AUTHENTICATOR_KEY.PASSWORD))) {
+        || (name === IDX_STEP.CHALLENGE_AUTHENTICATOR
+          && authenticatorKey === AUTHENTICATOR_KEY.PASSWORD))) {
```

Comment:
Yes sir, apparently this never worked correctly (mostly because it could not be tested since the Password Input component didn't have the implementation). But after adding the implementation and test it is working correctly with this condition.